boot.c

```diff
@@ -166,6 +161,19 @@ init_freemem()
   spa_init(freemem_va_start, freemem_size);
 }
 
+/* initialize user stack */
+void
+init_user_stack()
```

Comment:
Can this function ever fail?

Reply:
boot functions does not fail, and if it fails there's no fall back so it should just exit.
I'll add some assertions